### ISSUE TITLE
Tags repeatedly marked as not being set correctly for aws_db_instance

### ISSUE THREAD

**Issue by hashibot** [terraform-providers/terraform-provider-aws#345]
_This issue was originally opened by @davedash as hashicorp/terraform#9821. It was migrated here as part of the [provider split](https://www.hashicorp.com/blog/upcoming-provider-changes-in-terraform-0-10/). The original body of the issue is below._

<hr>

In Terraform 0.7.8 (and prior):

terraform plan gives me:

```
~ module.dev.aws_db_instance.db
    tags.%:      "0" => "3"
    tags.Name:   "" => "dbdev"
    tags.dbrole: "" => "master"
    tags.env:    "" => "dev"
```

My tf file looks like:

```

resource "aws_db_instance" "db" {
    ...
    engine               = "postgres"
    engine_version       = "9.5.2"
    instance_class       = "db.m3.medium"
    db_subnet_group_name = "${aws_db_subnet_group.db.name}"
    storage_encrypted    = true
    parameter_group_name = "${var.parameter_group}"  # summit custom
    allow_major_version_upgrade = true

    tags {
      Name = "${var.env}"
      dbrole = "master"
      env = "${var.env}"
    }
...
```

After an apply, the tags are set.  Re-running plan gives me the same output, so I'm always in a dirty state.

This is the case for all of my rds instances.

**Comment by bflad** [comment]
Courtesy of #4943, the `aws_db_instance` resource will properly return any `ListTagsForResource` error (our normal behavior for other parts of Terraform) instead of just a debug log and perpetual difference in `terraform plan`. It will be released in version 1.25.0 of the AWS provider, likely middle of next week.

**Comment by bflad** [comment]
This has been released in [version 1.25.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1250-june-27-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
